flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/ExecNodeGraphInternalPlan.java

```diff
@@ -75,6 +75,9 @@ public void writeToFile(File file, boolean ignoreIfExists, boolean failIfExists)
             }
         }
         try {
+            if (!file.getParentFile().exists()) {
```

Comment:
I don't think it's appropriate to make parent dirs here. The logic spreads across multiple classes...